src/include/migraphx/op/split.hpp

```diff
@@ -0,0 +1,210 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_SPLIT_HPP
+#define MIGRAPHX_GUARD_OPERATORS_SPLIT_HPP
+
+#include <array>
+#include <migraphx/operation.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/streamutils.hpp>
+#include <migraphx/literal.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/config.hpp>
+#include <cmath>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct split
+{
+    int axis = 0;
+    std::vector<int> slice_dims;
+    std::pair<int, int> slice_selector = {-1, -1};
+    std::string name() const { return "split"; }
+
+    std::vector<unsigned> compute_slice_elements(shape input_shape) const
+    {
+        unsigned accum = 1;
+        int axis_id    = 0;
+
+        //  compute accumulated elements on un-splitted axises.
+        for(auto&& len : input_shape.lens())
+        {
+            if(axis_id != axis)
+                accum *= len;
+            axis_id++;
+        }
+
+        // compute number of elements for each slice.
+        std::vector<unsigned> slice_elements;
+        std::transform(slice_dims.begin(),
+                       slice_dims.end(),
+                       std::back_inserter(slice_elements),
+                       [&](auto&& d) -> unsigned { return accum * d; });
+        return slice_elements;
+    }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        // check_shapes{inputs, *this}.has(1);
+        auto input_shape = inputs[0];
+        std::vector<std::size_t> out_dims;
+
+        if(slice_selector.first >= 0)
+        {
+            int first  = slice_selector.first;
+            int second = slice_selector.second;
+            if(second < first)
```

Comment:
Can we add the name of the operator to the err msg like "SPLIT: illegal split selector"?

Reply:
This is fixed.